### PR TITLE
RechartsWrapper event tests, and role=application

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Adding a couple of tests before I go ahead and mess with the implementation. I will try to add also tests for the payloads once I make sense of what's going on.

I also followed up on @julianna-langston advice from https://github.com/recharts/recharts/pull/4545#issuecomment-2119293141 and I added a role='application' to all charts.

## Related Issue

https://github.com/recharts/recharts/issues/4549

## Motivation and Context

tests and a11y

## How Has This Been Tested?

npm test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: src/chart/SunburstChart.tsx [hunk 1]
```diff
@@ -211,10 +212,11 @@ export const SunburstChart = ({
...
         <ViewBoxContext.Provider value={viewBox}>
           <TooltipContextProvider value={getTooltipContext()}>
-            <div
-              className={clsx('recharts-wrapper', className)}
-              style={{ position: 'relative', width, height }}
-              role="region"
+            <RechartsWrapper
+              className={className}
+              width={width}
+              // Sunburst doesn't support `style` property, why?
```
Comment (ckifer) [inline src/chart/SunburstChart.tsx:218]: I'm sure it could, it's the newest chart

File: test/cartesian/Area.spec.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ describe.each(chartsThatSupportArea)('<Area /> as a child of $testName', ({ Char
       };
 
       render(
-        <ChartElement width={500} height={500} data={data}>
```
Comment (ckifer) [inline test/cartesian/Area.spec.tsx:61]: Why are we getting rid of height and width? I'm missing something

**Review by ckifer — APPROVED** [review]
LGTM!

File: test/cartesian/Area.spec.tsx (hunk 2 above)
Comment (PavelVanecek) [reply test/cartesian/Area.spec.tsx:61]: I moved these to the test case definition because I didn't want to repeat it again.

Comment (ckifer) [reply test/cartesian/Area.spec.tsx:61]: Ah thanks

**Pull request merged**